pkg/workflows/sdk/v2/promise.go

```diff
@@ -0,0 +1,58 @@
+package sdk
+
+import (
+	"sync"
+)
+
+type Promise[T any] interface {
+	Await() (T, error)
+	CapabilityPromise
+}
+
+type CapabilityPromise interface {
+	promise()
+}
```

Comment:
Why is it important to prevent user created `Promise`s?

If it is not strictly necessary, then this sort of flexibility would be nice:
```suggestion
type Promise[T any] func() (T, error)
```